test/unit/model/test_objective.py

```diff
@@ -2,36 +2,128 @@
 #
 # SPDX-License-Identifier: MPL-2.0
 import unittest
-from test.utils.base import BaseTestCase
-from unittest.mock import MagicMock, patch
+import optuna
+from test.utils import BaseTestCase, TestData
+from datetime import datetime
 
-from openstf.enums import MLModelType
-from openstf.model.objective import XGBRegressorObjective
+from openstf.feature_engineering.feature_applicator import TrainFeatureApplicator
+from openstf.model.objective_creator import ObjectiveCreator
+from openstf.model.model_creator import ModelCreator
+from openstf.pipeline.optimize_hyperparameters import get_relevant_model_paramspace
+
+input_data = TestData.load("reference_sets/307-train-data.csv")
+input_data_with_features = TrainFeatureApplicator(horizons=[0.25, 24.0]).add_features(
+    input_data
+)
+input_data_with_features = input_data_with_features.iloc[::50, :]
```

Comment:
why select ::50? I would think you would use the whole train dataset

Reply:
It depends on the size of the train set I guess. Our unit test are already really slow and optimization with a large set could add a lot of time perhaps? I would say the goal of the test is to run though all paths of the code and not so much to actually deliver the best optimization on a large set.